### PR TITLE
Migrate docker base image to alpine for gateway and microservices

### PR DESCRIPTION
**Pull request by pascalgrimaud**
- related to https://github.com/jhipster/generator-jhipster/issues/2988
- migrate base image from `java:openjdk-8u72-jre` to `alpine:3.3`
- remove noregistry docker-compose files, after discussion here https://github.com/jhipster/generator-jhipster/pull/3029
- start microservice with registry:
    - `docker-compose -f src/main/docker/registy.yml -f src/main.docker/app.prod.yml up`
- start microservice (without registry because it's already started):
    - `docker-compose -f src/main/docker/app.prod.yml up`

Everything works fine, with scaling microservices.
jhipster-microservices

### CODE REVIEW

File: generators/server/templates/src/main/docker/_app.dev.yml [hunk 1]
```diff
@@ -72,5 +65,5 @@ services:
         <%_ } _%>
         <%_ if (applicationType == 'microservice') { _%>
         expose:
-            - "8080"
+            - "8081"
```
Comment (jd-carroll) [inline generators/server/templates/src/main/docker/_app.dev.yml]: Please forgive me, but I am confused by this... 
Why are you moving to port 8081, did I miss part of a thread somewhere?  And if so, why aren't you moving the [prod port](https://github.com/jhipster/generator-jhipster/pull/3085/files#diff-5e46f70be7afa0bd3a118c08d94e06b5L75) as well?

Comment (pascalgrimaud) [inline generators/server/templates/src/main/docker/_app.dev.yml]: Thanks for reviewing my docker PR, @jd-carroll, it's really appreciated!

About 8081 in dev profile : [application-dev.yml](https://github.com/jhipster/generator-jhipster/blob/master/generators/server/templates/src/main/resources/config/_application-dev.yml#L139)
We should probably do the same thing for prod profile, which is currently 8080 : [application-prod.yml](https://github.com/jhipster/generator-jhipster/blob/master/generators/server/templates/src/main/resources/config/_application-prod.yml#L136-L137)

Comment (deepu105) [inline generators/server/templates/src/main/docker/_app.dev.yml]: Can you plz add a todo item in our documentation ticket for these as well

Comment (pascalgrimaud) [inline generators/server/templates/src/main/docker/_app.dev.yml]: It is already the point 5 and 6, about docker updates and microservices

**Pull request merged**